pkg/ebpf/events/derived/derived.go

```diff
@@ -0,0 +1,29 @@
+package derived
+
+import "github.com/aquasecurity/tracee/types/trace"
+
+// DeriveFn is a function prototype for a function that receives an event as
+// argument and may produce a new event if relevant.
+// It returns list of derived events, which might be empty depending on successful derivation,
+// a bool indicating if an event was derived, and an error if one occurred.
+type DeriveFn func(trace.Event) ([]trace.Event, bool, error)
+
+// IDerivedEventGenerator is an interface for derived events generator class.
+// The main purpose of it is to create derived events given an event.
+// To generate derived event, the GenerateDerivedFn function should be used with initialized generator.
+type IDerivedEventGenerator interface {
+	GenerateEvents(event trace.Event) ([]trace.Event, bool, error)
+}
+
+func GenerateDerivedFn(generator IDerivedEventGenerator) DeriveFn {
```

Comment:
I don't see this being used anywhere, so what is the purpose of this function? Do you plan on using it in the future?

Reply:
Yea, I plan that each derived event implementation will use it for creating the current `DerivedFn`.

Reply:
So ditto to above, if we want to know if this will be more readable you need to write out an example to compare against.